polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -881,6 +892,18 @@ const isBreakpointsXS = () => {
         parseFloat(toPx(tokens.breakpoints['breakpoints-sm']) ?? '');
 };
 
+function getHeadingKey(heading: IndexTableHeading): string {
+  if ('id' in heading && heading.id) {
+    return heading.id;
+  }
+
```

Comment:
I cannot destructure heading because I must do some sort of inline type guard to check if `id` key is present.
<img width="400" alt="Screen Shot 2022-09-13 at 12 02 37 PM" src="https://user-images.githubusercontent.com/5626729/189954616-0a832976-23ec-45d0-8a42-923369f23a23.png">